### PR TITLE
--skip-npm in setup, don't build wheels, test commands, Re-rendered with conda-smithy 3.1.1 and pinning 2018.04.18 

### PR DESCRIPTION
**Pull request by bollwyvl**
Poking around https://github.com/jupyterlab/jupyterlab/issues/4418 it became pretty clear that something was up.

Local testing revealed that the old node installed on appveyor was being picked up (vs my clean VM that didn't have a system node). When there's no node, it installs just fine. If there _is_ node, and it fails, it fails hard. Adding `--install-options="--skip-npm"` Skips the node stuff

Also, the feedstock had missed a lot of the work in the last few months to get up to conda-build 3, which includes many fixes.

This also:
- updates the metadata
- pokes a number of the CLI commands in tests
- bump the build number

Basically all stuff we'll want to do anyway, even though it's possible that the `--skip-npm` might have been enough to get the windows build working.

:crossed_fingers: 

thanks to @afonit for the initial report and @jasongrout for the synchronized build debugging!

### CODE REVIEW

**Comment by bollwyvl** [comment]
Only failing build was win64/py27, which needs `subprocess32` for `jlpm`. Good to know!

That all looks good except for [this bit](https://ci.appveyor.com/project/conda-forge/jupyterlab-feedstock/build/1.0.236/job/bsylr06x4arjgw39#L962) on appveyor (also occurred on 3.6):
```
TEST END: C:\bld\win-64\jupyterlab-0.32.0-py35_1.tar.bz2
C:\bld\jupyterlab_1524106128836\_TEST_~1\Scripts\jupyter-migrate.exe - Access is denied.
C:\bld\jupyterlab_1524106128836\_TEST_~1\Scripts\jupyter-nbconvert.exe - Access is denied.
C:\bld\jupyterlab_1524106128836\_TEST_~1\Scripts\jupyter-troubleshoot.exe - Access is denied.
C:\bld\jupyterlab_1524106128836\_TEST_~1\Scripts\jupyter.exe - Access is denied.
# Automatic uploading is disabled
# If you want to upload package(s) to anaconda.org later, type:
anaconda upload C:\bld\win-64\jupyterlab-0.32.0-py35_1.tar.bz2
```

I guess that's when it's trying to uninstall?

**Comment by bollwyvl** [comment]
Woo all green on appveyor! Had to rekick a travis build due to network :japanese_ogre:, but it's nearly done now, too...

**Comment by jasongrout** [comment]
Do you want to add yourself as a maintainer too?

**Comment by jasongrout** [comment]
Thanks for working on this! I thought it was going to be a long night trying to diagnose these issues - what a pleasant surprise seeing it all done!

**Comment by jasongrout** [comment]
Thanks! And if you want to add yourself as a maintainer, please submit a PR - we'd love to have you on board.
